addons/account/models/account_move.py

```diff
@@ -1755,7 +1755,7 @@ def _get_unbalanced_moves(self, container):
         return self._cr.fetchall()
 
     def _check_fiscalyear_lock_date(self):
-        for move in self:
+        for move in self.filtered(lambda l: l.state == 'posted'):
```

Comment:
Doesn't this allow you to post a move with a date before the lock date?

Reply:
It does not, since there is another check on the lock date in the `_post` method. I could also add a parameter to this function to check only posted moves or all ... Not sure if it's currently worth the effort?

Reply:
But you could write `'state': 'posted'` manually without going through `_post`

Reply:
Since this is only for unlinking, wouldn't it make sense to filter only in `unlink`?

Reply:
Actually that makes total sense. I moved the check to the `unlink` method of `account.move.line`, which is being called in our use-case.